wgpu-hal/src/vulkan/device.rs

```diff
@@ -685,6 +686,162 @@ impl super::Device {
         }
     }
 
+    #[cfg(windows)]
+    fn find_memory_type_index(
+        &self,
+        type_bits: u32,
+        flags: vk::MemoryPropertyFlags,
+    ) -> Option<usize> {
+        let mem_properties = unsafe {
+            self.shared
+                .instance
+                .raw
+                .get_physical_device_memory_properties(self.shared.physical_device)
+        };
+
+        for (i, mem_ty) in mem_properties.memory_types_as_slice().iter().enumerate() {
+            if type_bits & (1 << i) != 0 && mem_ty.property_flags & flags == flags {
```

Comment:
Could we break up this expression into separate booleans with descriptive names/comments? I have no idea what this is doing just by looking at it.

Reply:
I referred to the `findProperties` function of https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkPhysicalDeviceMemoryProperties.html

Reply:
Done.